data-prepper-api/src/main/java/org/opensearch/dataprepper/model/types/ByteCount.java

```diff
@@ -79,11 +79,13 @@ public static ByteCount parse(final String string) {
         final String unitString = matcher.group("unit");
 
         if(unitString == null) {
-            throw new ByteCountInvalidInputException("Byte counts must have a unit.");
+            throw new ByteCountInvalidInputException("Byte counts must have a unit. Valid byte units include: " +
+                    Arrays.stream(Unit.values()).map(unitValue -> unitValue.unitString).collect(Collectors.toList()));
         }
 
         final Unit unit = Unit.fromString(unitString)
-                .orElseThrow(() -> new ByteCountInvalidInputException("Invalid byte unit: '" + unitString + "'"));
+                .orElseThrow(() -> new ByteCountInvalidInputException("Invalid byte unit: '" + unitString + "'. Valid byte units include: "
+                        + Arrays.stream(Unit.values()).map(unitValue -> unitValue.unitString).collect(Collectors.toList())));
```

Comment:
Probably a nit. This `Arrays.stream` statement repeated twice. It is also there in line number 83. I think you can add a method in Unit enum class to return this array of code values to keep the logic at one place.

Reply:
Line 51 already does something similar, though it creates a map which will be out of order. If you make it a linked map you could use the keys() from it.

https://github.com/opensearch-project/data-prepper/blob/c651aaae8896cf17f8973e558fbe8a98bced87f1/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/types/ByteCount.java#L51-L52